integrationTests/sync/edgeCases_test.go

```diff
@@ -0,0 +1,112 @@
+package sync
+
+import (
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/stretchr/testify/assert"
+)
+
+// TestSyncMetaNodeIsSyncingReceivedHigherRoundBlockFromShard tests the following scenario:
+// 1. Meta and shard 0 are in sync, producing blocks
+// 2. At nonce 3, shard 0 makes a rollback and stops producing blocks for 2 rounds, meta keeps producing blocks
+// 3. Shard 0 resumes creating blocks starting with nonce 3
+// 3. A bootstrapping meta node should be able to pass meta block with nonce 2
+func TestSyncMetaNodeIsSyncingReceivedHigherRoundBlockFromShard(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numNodesPerShard := 3
+	numNodesMeta := 3
+
+	nodes, advertiser, idxProposers := setupSyncNodesOneShardAndMeta(numNodesPerShard, numNodesMeta)
+	idxProposerMeta := idxProposers[1]
+	defer integrationTests.CloseProcessorNodes(nodes, advertiser)
+
+	integrationTests.StartP2pBootstrapOnProcessorNodes(nodes)
+	startSyncingBlocks(nodes)
+
+	round := uint64(0)
+	idxNonceShard := 0
+	idxNonceMeta := 1
+	nonces := []*uint64{new(uint64), new(uint64)}
+
+	round = integrationTests.IncrementAndPrintRound(round)
+	updateRound(nodes, round)
+	incrementNonces(nonces)
+
+	numRoundsBlocksAreProposedCorrectly := 3
+	proposeAndSyncBlocks(
+		nodes,
+		&round,
+		idxProposers,
+		nonces,
+		numRoundsBlocksAreProposedCorrectly,
+	)
+
+	shardIdToRollbackLastBlock := uint32(0)
+	manualRollback(nodes, shardIdToRollbackLastBlock, 3)
+	resetHighestProbableNonce(nodes, shardIdToRollbackLastBlock, 2)
+	emptyDataPools(nodes, shardIdToRollbackLastBlock)
+
+	//revert also the nonce, so the same block nonce will be used when shard will propose the next block
+	atomic.AddUint64(nonces[idxNonceShard], ^uint64(0))
```

Comment:
why initialization with max uint ?

Reply:
this is actually decrementing the nonce so it should be fine.

Reply:
Yes, that is from the original documentation of atomic.AddUint64